# <mayhem_program::instructions::pump_amm_interface::pump_swap_types::SwapPool as anchor_lang::AccountSerialize>::try_serialize  (yours 1520 B at .text+0x14fd8; original 1224 B at .text+0x80)
# distance 78 instructions; evidence: same-callees 2/2
--- original
+++ yours
@@ -4,13 +4,28 @@
 lddw r2, .rodata[f19a6d0411b16dbc]
 mov64 r3, 8
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+jeq r0, 0, +0xb0
+mov64 r1, r0
+and64 r1, 3
+mov64 r2, r1
+add64 r2, -2
+jlt r2, 2, +0x98
+jeq r1, 0, +0x98
+ldxdw r1, [r0+7]
+ldxdw r2, [r1]
+jeq r2, 0, +0x98
+ldxdw r1, [r0-1]
+callx r2
+mov64 r1, 3004
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::ErrorCode>>::from#ae9e
+ja +0x5d8
 mov64 r2, r7
 add64 r2, 274
 mov64 r1, r6
 mov64 r3, 1
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxh r1, [r7+272]
 stxh [r10-16], r1
 mov64 r2, r10
@@ -18,42 +33,49 @@
 mov64 r1, r6
 mov64 r3, 2
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 mov64 r1, r6
 mov64 r2, r7
 mov64 r3, 32
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 mov64 r2, r7
 add64 r2, 32
 mov64 r1, r6
 mov64 r3, 32
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 mov64 r2, r7
 add64 r2, 64
 mov64 r1, r6
 mov64 r3, 32
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 mov64 r2, r7
 add64 r2, 96
 mov64 r1, r6
 mov64 r3, 32
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 mov64 r2, r7
 add64 r2, 128
 mov64 r1, r6
 mov64 r3, 32
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 mov64 r2, r7
 add64 r2, 160
 mov64 r1, r6
 mov64 r3, 32
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxdw r1, [r7+240]
 stxdw [r10-16], r1
 mov64 r2, r10
@@ -61,13 +83,15 @@
 mov64 r1, r6
 mov64 r3, 8
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 mov64 r2, r7
 add64 r2, 192
 mov64 r1, r6
 mov64 r3, 32
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxb r1, [r7+275]
 stxb [r10-16], r1
 mov64 r2, r10
@@ -75,7 +99,8 @@
 mov64 r1, r6
 mov64 r3, 1
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxb r1, [r7+276]
 stxb [r10-16], r1
 mov64 r2, r10
@@ -83,7 +108,8 @@
 mov64 r1, r6
 mov64 r3, 1
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxdw r1, [r7+224]
 ldxdw r2, [r7+232]
 stxdw [r10-8], r2
@@ -93,7 +119,8 @@
 mov64 r1, r6
 mov64 r3, 16
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxdw r1, [r7+248]
 stxdw [r10-16], r1
 mov64 r2, r10
@@ -101,7 +128,8 @@
 mov64 r1, r6
 mov64 r3, 8
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxb r1, [r7+277]
 stxb [r10-16], r1
 mov64 r2, r10
@@ -109,7 +137,8 @@
 mov64 r1, r6
 mov64 r3, 1
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxb r1, [r7+278]
 stxb [r10-16], r1
 mov64 r2, r10
@@ -117,7 +146,8 @@
 mov64 r1, r6
 mov64 r3, 1
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxdw r1, [r7+256]
 stxdw [r10-16], r1
 mov64 r2, r10
@@ -125,7 +155,8 @@
 mov64 r1, r6
 mov64 r3, 8
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jne r0, 0, +0x448
+mov64 r8, r0
+jne r8, 0, +0x548
 ldxdw r1, [r7+264]
 stxdw [r10-16], r1
 mov64 r2, r10
@@ -133,20 +164,26 @@
 mov64 r1, r6
 mov64 r3, 8
 call <anchor_lang::bpf_writer::BpfWriter<&mut [u8]> as std::io::Write>::write_all#7274
-jeq r0, 0, +0x4b8
-mov64 r1, r0
-and64 r1, 3
-mov64 r2, r1
-add64 r2, -2
-jlt r2, 2, +0x4a0
-jeq r1, 0, +0x4a0
-ldxdw r1, [r0+7]
-ldxdw r2, [r1]
-jeq r2, 0, +0x4a0
-ldxdw r1, [r0-1]
-callx r2
+mov64 r8, r0
+jeq r8, 0, +0x5e0
 mov64 r1, 3004
 call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::ErrorCode>>::from#ae9e
+mov64 r2, r8
+and64 r2, 3
+mov64 r3, r2
+add64 r3, -2
+jlt r3, 2, +0x5d8
+jeq r2, 0, +0x5d8
+ldxdw r2, [r8+7]
+ldxdw r3, [r2]
+jeq r3, 0, +0x5d8
+ldxdw r2, [r8-1]
+mov64 r6, r1
+mov64 r1, r2
+mov64 r7, r0
+callx r3
+mov64 r1, r6
+mov64 r0, r7
 exit
 mov64 r0, 2
-ja +0x4b0
+ja +0x5d8
